Ensure filestore is shared when ran in a distributed way

lmignon opened the pull request:
When test are ran in a distributed way, we must ensure that each worker has access to the original filestore even if they uses a copy of the original database. This is done by patching the the filestore method of the odoo configuration to point to the original filestore and avoid to copy the original filestore for each created database.

This PR also provides a fix for tests using HTTP controllers.

lmignon committed:
Ensure filestore is shared when ran in a distributed way

When test are ran in a distributed way, we must ensure that each worker has access to the original filestore even if they uses a copy of the original database. This is done by patching the the filestore method of the odoo configuration to point to the original filestore and avoid to copy the original filestore for each created database

lmignon commented:
ping @yvaucher @petrus-v

sbidoul commented on pytest_odoo.py:
```diff
@@ -125,7 +141,8 @@ def _worker_db_name():
             os.system(f"dropdb {db_name} --if-exists")
             os.system(f"createdb -T {original_db_name} {db_name}")
```
```suggestion
            subprocess.run(["dropdb", db_name, "--if-exists"], check=True)
            subprocess.run(["createdb", "-T", original_db_name, db_name], check=True)
```

Not related to this PR but, @lmignon just in case the db name is not compatible with the shell separators.

lmignon replied:
good catch

lmignon committed:
Ensure dbfilter reference the database to use for tests

lmignon committed:
Uses subprocess lib to call shell command

This will prevent error in case the db name is not compatible with the shell separators

petrus-v commented:
good one ! thanks :+1:
